src/ui/pages/SidePage/components/IncomingRequest/IncomingRequest.tsx

```diff
@@ -1,54 +1,40 @@
 import { useEffect, useState, useMemo } from "react";
 import "./IncomingRequest.scss";
-import { Agent } from "../../../../../core/agent/agent";
 import { useAppDispatch, useAppSelector } from "../../../../../store/hooks";
 import {
-  dequeueCredentialRequest,
+  dequeueIncomingRequest,
   getQueueIncomingRequest,
 } from "../../../../../store/reducers/stateCache";
 import { SidePageContentProps } from "../../SidePage.types";
-import { RequestComponent } from "./components/RequestComponent";
-import {
-  IncomingRequestProps,
-  IncomingRequestType,
-} from "../../../../../store/reducers/stateCache/stateCache.types";
-import { getConnectedWallet } from "../../../../../store/reducers/walletConnectionsCache";
+import { SignRequest } from "./components/SignRequest"; // Import SignRequest component
+import { PeerConnectSigningEventRequest } from "../../../../../store/reducers/stateCache/stateCache.types";
 
 const IncomingRequest = ({ open, setOpenPage }: SidePageContentProps) => {
   const pageId = "incoming-request";
   const dispatch = useAppDispatch();
   const queueIncomingRequest = useAppSelector(getQueueIncomingRequest);
-  const connectedWallet = useAppSelector(getConnectedWallet);
   const incomingRequest = useMemo(() => {
     if (
       !queueIncomingRequest.isProcessing ||
       !queueIncomingRequest.queues.length
     ) {
       return;
     } else {
-      return queueIncomingRequest.queues[0];
+      return queueIncomingRequest.queues[0] as PeerConnectSigningEventRequest;
     }
   }, [queueIncomingRequest]);
   const [initiateAnimation, setInitiateAnimation] = useState(false);
-  const [requestData, setRequestData] = useState<IncomingRequestProps>();
+  const [requestData, setRequestData] =
```

Comment:
My worry with the changes in this file are now we assume `IncomingRequest` is always a peer connect signing event, so if we need to extend it with any other actions in the future we have to reverse what this PR is doing. (e.g. removing the `RequestComponent` component)

Reply:
Yes, I know, but since we have no other use cases for now I thought it would be better to keep it simple. Adding that back won't be difficult anyway.

Reply:
I guess from a maintainability standard it's not obvious how it's extended, if this happens much later and it's not done by you (e.g. you are on a different project or something).

Reply:
It's really a simple thing to do... that request component was just a container with styles and a switch inside. All styles were also very specific to the cases that have now been removed. It was something that could have been done inside `IncomingRequest` but I decided to give it a dedicated file to make things cleaner.

Reply:
Alright, can we at least leave a comment in the code about extending? It may be simple to you now but we don't know how this code will develop or who will be making that change.